core/src/main/java/org/mule/internal/connection/DefaultConnectionManager.java

```diff
@@ -176,7 +178,8 @@ private void close(ConnectionHandlingStrategyAdapter managementStrategy)
 
     private <Config, Connection> ConnectionHandlingStrategyAdapter<Config, Connection> getManagementStrategy(Config config, ConnectionProvider<Config, Connection> connectionProvider)
     {
-        return (ConnectionHandlingStrategyAdapter<Config, Connection>) connectionProvider.getHandlingStrategy(new DefaultConnectionHandlingStrategyFactory(config, connectionProvider, muleContext));
+        PoolingProfile poolingProfile = ((ConnectionProviderWrapper) connectionProvider).getPoolingProfile().orElse(getDefaultPoolingProfile());
```

Comment:
don't cast blindly... add an instanceof condition and if else use the default. It'd also be good if you move that to a separate method